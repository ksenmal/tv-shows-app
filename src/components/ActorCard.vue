<script setup>
defineProps({
  actor: Object,
});
</script>

<template>
  <img
    v-if="
      actor.character && actor.character.image && actor.character.image.medium
    "
    :src="actor.character.image.medium"
    :alt="actor.character.name"
  />
  <div>
    <p v-if="actor.person" class="actor-name">{{ actor.person.name }}</p>
    <p v-if="actor.character">
      as <span class="character-name">{{ actor.character.name }}</span>
    </p>
  </div>
</template>

<style scoped>
img {
  width: 150px;
  height: auto;
  margin-right: 1rem;
}
.actor-name {
  font-size: 20px;
}
.character-name {
  font-weight: bold;
}
</style>
