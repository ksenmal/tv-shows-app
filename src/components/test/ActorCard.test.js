import { shallowMount } from "@vue/test-utils";
import ActorCard from "../ActorCard.vue";
import { describe, expect, test } from "vitest";

describe("ActorCard suite", () => {
  test("Snapshot", () => {
    const wrapper = shallowMount(ActorCard, {
      props: {
        actor: {
          character: {
            image: {
              medium: "https://example.com/img.jpg",
            },
            name: "Jon Snow",
          },
          person: {
            image: {
              medium: "https://example.com/img.jpg",
            },
            name: "Kit Harington",
          },
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
