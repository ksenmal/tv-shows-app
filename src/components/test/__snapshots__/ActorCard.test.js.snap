// Vitest Snapshot v1

exports[`ActorCard suite > Snapshot 1`] = `
"<img src=\\"https://example.com/img.jpg\\" alt=\\"Jon Snow\\" data-v-b810ceff=\\"\\">
<div data-v-b810ceff=\\"\\">
  <p class=\\"actor-name\\" data-v-b810ceff=\\"\\">Kit Harington</p>
  <p data-v-b810ceff=\\"\\"> as <span class=\\"character-name\\" data-v-b810ceff=\\"\\">Jon Snow</span></p>
</div>"
`;
